openedx/core/djangoapps/util/ip.py

```diff
@@ -0,0 +1,186 @@
+"""
+Utilities for determining the IP address of a request.
+
+``get_client_ip`` is the main function of interest.
+"""
+
+import ipaddress
+import warnings
+
+from django.conf import settings
+
+
+def get_meta_ips(request_meta, header_name):
+    """
+    Return the list of IPs the request is carrying on this header, which is
+    expected to be comma-delimited if it contains more than one. Response
+    may be an empty list for missing or empty header. List items may not be
+    valid IPs.
+    """
+    if not header_name:
+        return []
+
+    field_name = 'HTTP_' + header_name.replace('-', '_').upper()
+    header_value = request_meta.get(field_name, '').strip()
+
+    if header_value:
+        return [s.strip() for s in header_value.split(',')]
+    else:
+        return []
+
+
+def get_client_ip_by_one_header(request_meta, header_name, index):
+    """
+    Implementation of ``get_client_ip_via_configured_header`` for one header.
+    Returns None if header name and index can't be used.
+    """
+    ip_strs = get_meta_ips(request_meta, header_name)
+
+    if not ip_strs:
+        warnings.warn(f"Configured IP address header was missing: {header_name!r}", UserWarning)
```

Comment:
The warnings are configured (by default? in edxapp? not sure) to only be printed the first time that exact warning string is issued. It will be logged, but just once. (Maybe just once for that module.) In any case, it shouldn't be noisy.

It should be issued at a warning level, since it means someone's settings are out of sync with their network configuration.

Reply:
Having multiple means you can make a transition from one network configuration to another, expand-contract style, without loss of IP-determination coverage.